### PR TITLE
STY: add `ruff-format` and remove `flake8`

### PR DESCRIPTION
**Pull request by paddyroddy**
The maintainer of `flake8` can be a bit difficult, so nice to remove. I will add `ruff` the linter in a separate PR, but would be nice to get formatting done first. `ruff-format` is basically faster `black` (but some subtle differences).

Reviewed-by: Saransh Chopra <saransh0701@gmail.com>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,12 +10,13 @@ on:
...
-    steps:
-    - uses: actions/checkout@v4
-    - run: pipx run flake8 glass
+# TODO: replace with a more general pre-commit linter
+#   style:
+#     name: Style
+#     runs-on: ubuntu-latest
+#     steps:
+#     - uses: actions/checkout@v4
+#     - run: pipx run flake8 glass
```
Comment (Saransh-cpp) [inline .github/workflows/test.yml:19]: Unrelated to this PR, but maybe we can add pre-commit.ci to the repository?

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
This will mess up [`git blame`](https://git-scm.com/docs/git-blame)'s output. A good practice is to add the commit where `ruff` made the formatting changes to a new `.git-blame-ignore-revs` file. For example, PyBaMM's `.git-blame-ignore-revs` file - https://github.com/pybamm-team/PyBaMM/blob/develop/.git-blame-ignore-revs

File: .github/workflows/test.yml (hunk 1 above)
Comment (paddyroddy) [reply .github/workflows/test.yml:19]: The problem is the changes invalidate `flake8` so it. kind of had to go

Comment (paddyroddy) [reply .github/workflows/test.yml:19]: I've never been a fan of the CI as it only works in public repos, but seeing as its public I guess we could

**Comment by paddyroddy** [comment]
> This will mess up [`git blame`](https://git-scm.com/docs/git-blame)'s output. A good practice is to add the commit where `ruff` made the formatting changes to a new `.git-blame-ignore-revs` file. For example, PyBaMM's `.git-blame-ignore-revs` file - [pybamm-team/PyBaMM@`develop`/.git-blame-ignore-revs](https://github.com/pybamm-team/PyBaMM/blob/develop/.git-blame-ignore-revs?rgh-link-date=2024-09-11T17%3A29%3A09Z)

Wasn't aware of this idea to combat `git blame`, have added the file

File: .github/workflows/test.yml (hunk 1 above)
Comment (Saransh-cpp) [reply .github/workflows/test.yml:19]: Also could use the pre-commit or the pre-commit-lite action here.

**Review by Saransh-cpp — APPROVED** [review]
Thanks!

File: .github/workflows/test.yml (hunk 1 above)
Comment (paddyroddy) [reply .github/workflows/test.yml:19]: Will do in a different PR

**Comment by ntessore** [comment]
The commit where ruff made the changes will get squashed, so we should add that file not here but in a PR immediately after.

Also, why not add the linter in this PR as well, so we only incur one massive changeset?

**Comment by Saransh-cpp** [comment]
> The commit where ruff made the changes will get squashed, so we should add that file not here but in a PR immediately after.

Ah, that too. This makes more sense then.

**Comment by paddyroddy** [comment]
> The commit where ruff made the changes will get squashed, so we should add that file not here but in a PR immediately after.

Yeah I wasn't sure on this

**Comment by paddyroddy** [comment]
> Also, why not add the linter in this PR as well, so we only incur one massive changeset?

Because formatting changes and linting changes are different. I think it will hide a lot more changes that way.

**Review by ntessore — APPROVED** [review]

**Comment by paddyroddy** [comment]
I have updated from `main`. I am going to merge, and then update the commit for the `git blame` file

**Comment by paddyroddy** [comment]
Have merged directly https://github.com/glass-dev/glass/commit/e01b40ec11278be4aa02d52c9ad76ce09bc19a6a

**Pull request merged**